geniza/corpus/views.py

```diff
@@ -84,17 +92,50 @@ class DocumentDetailView(DetailView):
 
     context_object_name = "document"
 
+    def page_title(self):
+        return self.get_object().title
+
+    def page_description(self):
+        return Truncator(self.get_object().description).words(20)
+
     def get_queryset(self, *args, **kwargs):
         """Don't show document if it isn't public"""
         queryset = super().get_queryset(*args, **kwargs)
         return queryset.filter(status=Document.PUBLIC)
 
+    def get_context_data(self, **kwargs):
+        context_data = super().get_context_data(**kwargs)
+        context_data.update(
+            {
+                "page_title": self.page_title(),
+                "page_description": self.page_description(),
+            }
+        )
+        return context_data
+
 
 class DocumentScholarshipView(DocumentDetailView):
     """List of :class:`~geniza.footnotes.models.Footnote`s for a Document"""
 
     template_name = "corpus/document_scholarship.html"
 
+    def page_title(self):
+        # Translators: title of document scholarship page
+        return _("Scholarship on %(doc)s") % {"doc": self.get_object().title}
+
+    def page_description(self):
+        doc = self.get_object()
+        count = doc.footnotes.count()
+        # Translators: description of document scholarship page, for search engines
+        return ngettext(
+            "%(count)d scholarship record for %(doc)s",
+            "%(count)d scholarship records for %(doc)s",
```

Comment:
is this repeated for singular/plural? worth a comment or just a convention we'll get used to once we use it more?

Reply:
It is, yeah. Maybe a convention we'll get used to; the docs explain more about it but I'm still figuring out how to do it correctly in templates